cji_smoke_test.sh

```diff
@@ -162,9 +162,10 @@ mc --no-color --quiet mirror --overwrite minio/${BUCKET_NAME} /artifacts/
 "
 
 run_mc () {
-    echo "running: docker run -t --net=host --name=$CONTAINER_NAME --entrypoint=\"/bin/sh\" $MC_IMAGE -c \"$CMD\""
+    # image was already pulled earlier in this script, use --pull=never
+    echo "running: docker run -t --net=bridge --pull=never --name=$CONTAINER_NAME --entrypoint=\"/bin/sh\" $MC_IMAGE -c \"$CMD\""
     set +e
-    docker run -t --net=host --name=$CONTAINER_NAME --entrypoint="/bin/sh" $MC_IMAGE -c "$CMD"
+    docker run -t --net=bridge --pull=never --name=$CONTAINER_NAME --entrypoint="/bin/sh" $MC_IMAGE -c "$CMD"
```

Comment:
how about not specifying the pull policy at all? from all three, `missing`, `never`, `always` , `missing` is the default one, I think we'd need to pull it at some point when the new nodes have no imaged cached at all.

I think we chose `always` so that the `latest` image tag served an actual purpose - but again, on newer nodes the images will have to be pulled at least once.

Reply:
We actually pull it here: https://github.com/RedHatInsights/cicd-tools/pull/104/files#diff-10065b743d1089868450d50904d5d59c03b0a255158e1543e183b220a923bbdeR56

Which is why I am trying to see if adding 'pull=never' helps with our problem. docker/podman should have no need to check the registry again for this image... but for whatever reason, it seems like it is and then sometimes the script hits the 'unauthorized' error. But that first 'docker pull' always seems to pass.

Reply:
Sure, I did not see the pull before that, I guess I wasn't checking the whole file and just looking at this change.

I do think the problem seems related to something deleting mid-run the credentials file (from another process, causing the pull to fail) - sure if you are able to pull and then don't reach the registry again this should help - one would argue why you do the pull "in advance" but that's a different story. 

I guess this mitigates the point of failure by reducing the interaction with the registry - still we don't know what credentials this process might be using at that point ... perhaps might be worthy to try and echo the value of `DOCKER_CONFIG`?